### ISSUE TITLE
selector-first-class-pattern

### ISSUE THREAD

**Issue by DoroNahari** [stylelint/stylelint#3209]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Specify the first pattern for class selectors that should or shouldn't use.

> Which rule, if any, is this issue related to?

`selector-class-pattern`

> What CSS is needed to reproduce this issue?

```css
.foo .goo , .foo {

}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
		"selector-first-class-pattern": ["^(?!foo).*$", {
			"severity": "error"
		}]
  }
}
```

> Which version of stylelint are you using?

`9.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

 `Node API`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following errors were flagged:

error source="stylelint.rules.selector-first-class-pattern" line="1" column="2" severity="error" message="Class selector "foo"; can not be the first class (selector-first-class-pattern)

error source="stylelint.rules.selector-first-class-pattern" line="1" column="14" severity="error" message="Class selector "foo"; can not be the first class (selector-first-class-pattern)

**Comment by jeddy3** [comment]
@DoroNahari Thanks for the request and for using the template. I'm having a little trouble groking your use case from the example. Are you able to flesh it out a little, please?

**Comment by DoroNahari** [comment]
I have updated the example.
In general the rule target is to block peoples from override designs by blocking the option of put a specific class at the begginning of the declaring

**Comment by jeddy3** [comment]
@DoroNahari Thanks for updating the example. I'm not sure this proposal meets the [criteria for inclusion](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion) as it's a rather idiosyncratic pattern. Fear not, stylelint has a [plugin system](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) for this very purpose. You can create and tailor your own plugin to your specific needs. You can look at the [existing plugins](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/plugins.md) for implementation ideas. Good luck!

**Comment by DoroNahari** [comment]
If i build this plugin, will you consider adding it to Stylelint?
